TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodyAccumulateHandler.java

```diff
@@ -22,30 +27,62 @@ public class NettyJsonBodyAccumulateHandler extends ChannelInboundHandlerAdapter
+        // For this buffer to continue to be used, we are far-off the happy-path.
+        // Consolidating will likely burn more cycles
+        this.accumulatedBody = Unpooled.compositeBuffer(1024);
+        this.parsedJsonObjects = new ArrayList<>();
     }
 
     @Override
     public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
         if (msg instanceof HttpJsonMessageWithFaultingPayload) {
             capturedHttpJsonMessage = (HttpJsonMessageWithFaultingPayload) msg;
         } else if (msg instanceof HttpContent) {
-            var jsonObject = jsonAccumulator.consumeByteBuffer(((HttpContent) msg).content().nioBuffer());
-            if (jsonObject != null) {
-                capturedHttpJsonMessage.payload()
-                    .put(JsonKeysForHttpMessage.INLINED_JSON_BODY_DOCUMENT_KEY, jsonObject);
-                context.onJsonPayloadParseSucceeded();
+            var contentBuf = ((HttpContent) msg).content();
+            accumulatedBody.addComponent(true, contentBuf.retainedDuplicate());
+            var nioBuf = contentBuf.nioBuffer();
+            jsonAccumulator.consumeByteBuffer(nioBuf);
+            Object nextObj;
+            while ((nextObj = jsonAccumulator.getNextTopLevelObject()) != null) {
+                parsedJsonObjects.add(nextObj);
+            }
+            if (msg instanceof LastHttpContent) {
+                if (!parsedJsonObjects.isEmpty()) {
+                    var payload = capturedHttpJsonMessage.payload();
+                    if (parsedJsonObjects.size() > 1) {
+                        payload.put(JsonKeysForHttpMessage.INLINED_NDJSON_BODIES_DOCUMENT_KEY, parsedJsonObjects);
+                    } else {
+                        payload.put(JsonKeysForHttpMessage.INLINED_JSON_BODY_DOCUMENT_KEY, parsedJsonObjects.get(0));
+                    }
+                    if (!jsonAccumulator.hasPartialValues()) {
+                        context.onJsonPayloadParseSucceeded();
+                    }
+                }
+                if (jsonAccumulator.hasPartialValues() || parsedJsonObjects.isEmpty()) {
+                    if (jsonAccumulator.getTotalBytesFullyConsumed() > Integer.MAX_VALUE) {
+                        throw new IndexOutOfBoundsException("JSON contents were too large " +
+                            jsonAccumulator.getTotalBytesFullyConsumed() + " for a single composite ByteBuf");
+                    }
+                    // skip the contents that were already parsed and included in the payload as parsed json
+                    accumulatedBody.readerIndex((int) jsonAccumulator.getTotalBytesFullyConsumed());
+                    // and pass the remaining stream
+                    capturedHttpJsonMessage.payload()
+                        .put(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY, accumulatedBody);
+                } else {
+                    accumulatedBody.release();
```

Comment:
Can we also do a release if the channel is unregistered and accumulatedBody is still present